### PR TITLE
[CELEBORN-1058] Support specifying the number of dispatcher threads for each role

### PR DESCRIPTION
**Pull request by onebox-li**
### What changes were proposed in this pull request?
Support specifying the number of dispatcher threads for each role, especially shuffle client side. For shuffle client, there is only RpcEndpointVerifier endpoint which handles not many requests, one thread is enough. The rpc env of other roles has only two endpoints at most, using a shared event loop is reasonable. I am not sure if there is a need to add rpc requests to shuffle client. So add specific parameters to specify the dispatcher threads here.

And change the dispatcher thread pool name in order to distinguish it from spark's.

### Why are the changes needed?
Ditto

### Does this PR introduce _any_ user-facing change?
Yes, add params celeborn.\<role>.rpc.dispatcher.threads

### How was this patch tested?
Manual test and UT

### CODE REVIEW

File: common/src/main/java/org/apache/celeborn/common/protocol/RpcNameConstants.java [hunk 1]
```diff
@@ -19,18 +19,19 @@
...
 public class RpcNameConstants {
   // For Master
-  public static String MASTER_SYS = "MasterSys";
-
+  public static String MASTER_SYS = "Master";
   // Master Endpoint Name
   public static String MASTER_EP = "MasterEndpoint";
 
   // For Worker
-  public static String WORKER_SYS = "WorkerSys";
```
Comment (FMX) [inline common/src/main/java/org/apache/celeborn/common/protocol/RpcNameConstants.java:28]: What will happen if the server is using this PR and the client is old? Will this name change introduce a compatible issue?

Comment (onebox-li) [reply common/src/main/java/org/apache/celeborn/common/protocol/RpcNameConstants.java:28]: The xx_SYS is just used for startup logs before this pr. I think it doesn't matter.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -381,8 +381,17 @@ class CelebornConf(loadDefaults: Boolean) extends Cloneable with Logging with Se
     new RpcTimeout(get(RPC_LOOKUP_TIMEOUT).milli, RPC_LOOKUP_TIMEOUT.key)
   def rpcAskTimeout: RpcTimeout =
     new RpcTimeout(get(RPC_ASK_TIMEOUT).milli, RPC_ASK_TIMEOUT.key)
-  def rpcDispatcherNumThreads(availableCores: Int): Int =
-    get(RPC_DISPATCHER_THREADS).getOrElse(availableCores)
+  def rpcDispatcherNumThreads(availableCores: Int): Int = {
+    val num = get(RPC_DISPATCHER_THREADS)
```
Comment (SteNicholas) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:385]: Could this method invoke `rpcDispatcherNumThreads(availableCores: Int, role: String)`?

Comment (onebox-li) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:385]: I think it can’t. It is a fallback condition in this pr.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main(v0.4.0)/branch-0.3(v0.3.2)

**Comment by waitinfuture** [comment]
Hi @onebox-li , do you have recommended dispatcher threads for different roles? We can add them to the document and config template.

**Comment by onebox-li** [comment]
> Hi @onebox-li , do you have recommended dispatcher threads for different roles? We can add them to the document and config template.

Thanks @waitinfuture for suggestion. I think what is certain now is that one thread for the shuffle client should be enough. The number of dispatcher threads for other roles depends on the user's RSS cluster scale or task size and default configuration may be more universal. I thought about this before, and I will explore if there are any good recommended configuration suggestion for this.

**Comment by waitinfuture** [comment]
> > Hi @onebox-li , do you have recommended dispatcher threads for different roles? We can add them to the document and config template.
> 
> Thanks @waitinfuture for suggestion. I think what is certain now is that one thread for the shuffle client should be enough. The number of dispatcher threads for other roles depends on the user's RSS cluster scale or task size and default configuration may be more universal. I thought about this before, and I will explore if there are any good recommended configuration suggestion for this.

Thanks, so at least we should add the client side config to the template, and also add the recommendation in the config's document.

**Pull request closed without merging**